core/src/main/java/org/apache/iceberg/ManifestReader.java

```diff
@@ -289,12 +296,12 @@ static boolean dropStats(Expression rowFilter, Collection<String> columns) {
         Sets.intersection(Sets.newHashSet(columns), STATS_COLUMNS).isEmpty();
   }
 
-  private static Collection<String> withStatsColumns(Collection<String> columns) {
+  static Collection<String> withStatsColumns(Collection<String> columns) {
```

Comment:
If this returned `List`, then we wouldn't need to copy the list in `ManifestGroup`

Reply:
In one of the usage `columns` directly comes from [the field `column` which is a collection](https://github.com/apache/iceberg/blob/b1296bcbe8e050d4bc28e3d41feb2f8868c8f0bf/core/src/main/java/org/apache/iceberg/ManifestReader.java#L84) so I didn't change that, but it seems that from all its current usage this field could be changed to a list. I'll need to update a few tests after this but I think it's doable. Will do!